dvc/utils/strictyaml.py

```diff
@@ -101,25 +125,148 @@ def prepare_code(mark: "StreamMark") -> "Syntax":
             if exc.problem_mark is not None:
                 lines.append(prepare_code(exc.problem_mark))
 
-        if lines and lines[-1]:
+        if lines:
+            # we should not add a newline after the main message
+            # if there are no other outputs
             lines.insert(0, "")
-        lines.insert(0, f"[red]'{relpath(self.path)}' structure is corrupted.")
-        for message in lines:
-            ui.error_write(message, styled=True)
 
+        rel = make_relpath(self.path)
+        message = f"'{rel}' is corrupted{self.hint}"
+        lines.insert(0, f"[red]{message}.")
+        for line in lines:
+            ui.error_write(line, styled=True)
+
+
+def determine_linecol(
```

Comment:
Line numbers for schema validation errors may be inaccurate sometimes. Please read the doc below for more technical information.